### PR TITLE
[Tech Debt] Replace deprecated wait() with waitFor()

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves N/A

**Overall change:**
testing-library-react has deprecated the `wait()` method and replaced it with `waitFor()` with a now required callback.

https://testing-library.com/docs/dom-testing-library/api-async#wait-deprecated-use-waitfor-instead

image

This PR replaces all occurrences of `wait()` with waitFor()

**Code changes:**

- Replace `wait()` with `waitFor()`

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/FrontPage/index.test.jsx [hunk 1]
```diff
@@ -163,7 +163,7 @@ describe('Front Page', () => {
       expect(langSpan.getAttribute('lang')).toEqual('en-GB');
       expect(langSpan.textContent).toEqual('BBC News');
 
-      await wait();
+      await waitFor(() => {});
```
Comment (chris-hinds) [inline src/app/pages/FrontPage/index.test.jsx:166]: `waitFor()` now requires a callback even if it is empty.

**Review by jroebu14 — APPROVED** [review]

File: src/app/pages/FrontPage/index.test.jsx (hunk 1 above)
Comment (DenisHdz) [reply src/app/pages/FrontPage/index.test.jsx:166]: Could you check if these empty `wait`s are really necessary?

Comment (chris-hinds) [reply src/app/pages/FrontPage/index.test.jsx:166]: will double check

Comment (chris-hinds) [reply src/app/pages/FrontPage/index.test.jsx:166]: The wait is needed here due to some async updates that happen within the toggles containers. A future PR could refactor this to use `act(() =>{})` to handle those updates.

**Review by simonsinclair — APPROVED** [review]

File: src/app/pages/FrontPage/index.test.jsx (hunk 1 above)
Comment (sareh) [reply src/app/pages/FrontPage/index.test.jsx:166]: @hindsc52 Thanks for this PR. Could you please write up an issue to move from using waitFor() to act() for all relevant tests in the codebase?

**Review by sareh — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]

File: src/app/pages/FrontPage/index.test.jsx (hunk 1 above)
Comment (chris-hinds) [reply src/app/pages/FrontPage/index.test.jsx:166]: https://github.com/bbc/simorgh/issues/6384

**Pull request merged**